lib/grpcApi/handlers/commitTransactionHandlerFactory.js

```diff
@@ -0,0 +1,64 @@
+const { CommitTransactionResponse } = require('@dashevo/drive-grpc');
+
+const InvalidArgumentGrpcError = require('../error/InvalidArgumentGrpcError');
+const InternalGrpcError = require('../error/InternalGrpcError');
+
+/**
+ * @param {MongoDBTransaction} stateViewTransaction
+ * @param {SVContractMongoDbRepository} svContractMongoDbRepository
+ * @param {createContractDatabase} createContractDatabase
+ * @param {removeContractDatabase} removeContractDatabase
+ * @returns {commitTransactionHandler}
+ */
+module.exports = function commitTransactionHandlerFactory(
+  stateViewTransaction,
+  svContractMongoDbRepository,
+  createContractDatabase,
+  removeContractDatabase,
+) {
+  /**
+   * Commit transaction, that was created before and create collections for
+   * documents inside of received contracts
+   *
+   * @typedef commitTransactionHandler
+   * @param {Object} call
+   * @returns {Promise<CommitTransactionResponse>}
+   */
+  async function commitTransactionHandler(call) {
+    const { request } = call;
+
+    const blockHash = request.getBlockHash();
+
+    if (!blockHash) {
+      throw new InvalidArgumentGrpcError('blockHash is not specified');
+    }
+
+    const contracts = await svContractMongoDbRepository
```

Comment:
We can avoid of this DB query if we store contracts for the current state view transaction in memory.